src/clj_telnet/core.clj

```diff
@@ -31,6 +34,26 @@
   ([^String server-ip]
    (get-telnet server-ip 23)))
 
+(defn  print-c-debug [c]
+  (when *debug* (print (char c))))
+
+(defn print-data-debug [data]
+  (when *debug* (print data)))
+
+(defn- read-a-char
+  [in]
+  (try
+    (let [c (.read in)]
+      (print-c-debug c)
+      c)
+    (catch SocketTimeoutException ste (println "Read error") nil)))
```

Comment:
I am concerned here. This would catch an exception not previously caught. That could break backwards behavior.

Reply:
Now I would be a fan of using a dynamic function to behave as previously. That would give the user the option to bind the function to something that throws an error.
```
(defn ^:dynamic read-err [e]
  (throw e))
```
Then if someone wanted to swallow the exception and log something, they could.
So together you'd have something like...

```
(defn ^:dynamic read-err [e] (throw e))
(defn- read-a-char  [in]
  (try
    (let [c (.read in)]
      (print-c-debug c)
      c)
    (catch SocketTimeoutException ste (read-err ste)))))
```